### PR TITLE
Move pilot-hubs user image to its own repo

### PR DESCRIPTION
**Pull request by GeorgianaElena**
### This PR:
- removes the user image files
- removes the build script and references to it
- updates docs on how to update default user environment

### Not in this PR:
- New pilot-hubs-image repository that uses repo2docker action is :arrow_right: ~https://github.com/2i2c-org/pilot-hubs-image~ https://github.com/2i2c-org/2i2c-hubs-image
- Images are pushed to this quay.io registry :arrow_right: ~https://quay.io/repository/2i2c/pilot-hubs-image?tab=info~ https://quay.io/repository/2i2c/2i2c-hubs-image?tab=info

Closes https://github.com/2i2c-org/infrastructure/issues/710

### CODE REVIEW

**Comment by sgibson91** [comment]
The only question I had was about calling the new repo `pilot-hubs-image`. Is this not confusing/misleading since the `pilot-hubs` repo is now called `infrastructure`?

**Comment by GeorgianaElena** [comment]
You're totally right @sgibson91! I thought about this too and what made me go with this name was:
- `infrastructure-image` didn't feel right as it doesn't give any initial info about the image
- I believe we're still calling the hubs deployed from this repo pilot hubs or community hubs throughout our docs

**Comment by sgibson91** [comment]
Thanks for explaining @GeorgianaElena! This sounds reasonable to me. We can always update it later if/when we decide to drop the "pilots" prefix :)

**Review by sgibson91 — APPROVED** [review]

**Review by choldgraf — COMMENTED** [review]
This looks good to me! Though I actually think we should avoid using the word `pilot` for now. How about we just go with `2i2c-hubs-image` now?

My rationale here is that I've gotten some feedback from others that it can be hard to advocate internally for using 2i2c's services when we use words like "pilot", "alpha", etc. It makes decision-makers think the infrastructure is "not ready for users", which I don't think is the impression we want to give, so I think we should move away from that language.

I'm +1 other than this suggestion though

**Comment by GeorgianaElena** [comment]
I've updated the GitHub and Quay repositories name to be `2i2c-hubs-image`! Thanks @sgibson91 and @choldgraf for the feedback.

File: docs/howto/configure/update-env.md [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
 
-1. Install pre-requisites:
-   * Docker
-   * A Python virtual environment. Install `requirements.txt` into it.
-   * The `gcloud` tool, authenticated to the `two-eye-two-see` project.
-      You need to run `gcloud auth configure-docker us-central1-docker.pkg.dev`
-      once as well.
+The image is built and pushed using [jupyterhub/repo2docker-action](https://github.com/jupyterhub/repo2docker-action)
+to the [pilot-hubs-image quay.io](https://quay.io/repository/2i2c/pilot-hubs-image).
+registry
```
Comment (choldgraf) [inline docs/howto/configure/update-env.md:8]:
```suggestion
The image is built and pushed using [jupyterhub/repo2docker-action](https://github.com/jupyterhub/repo2docker-action) to the [pilot-hubs-image quay.io](https://quay.io/repository/2i2c/pilot-hubs-image) registry.
```

File: docs/howto/configure/update-env.md [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 
-2. Make the changes you need to make to the environment, and git commit it.
-
-3. Run `python3 deployer build <cluster-name>`, once for each `<cluster>` under `config/hubs`.
-   This will build the image and push it to registry. It will tell you what the generated image tag is.
+To update this environment:
 
+1. Make the changes you need to make to the environment, by editing the files in
+   the [pilot-hubs-image repository](https://github.com/2i2c-org/pilot-hubs-image)
```
Comment (choldgraf) [inline docs/howto/configure/update-env.md:13]:
```suggestion
   the [2i2c-hubs-image repository](https://github.com/2i2c-org/2i2c-hubs-image)
```

File: docs/howto/configure/update-env.md [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
+   and open a pull request. This will also provide you the ability to test your
+   changes on Binder to make sure everything works as expected.
+2. Once the PR is merged or a commit to the `main` branch happens, the new image will be pushed
+   to the registry.
+3. Get the latest tag from the [tags section](https://quay.io/repository/2i2c/pilot-hubs-image?tab=tags) in quay.io
```
Comment (choldgraf) [inline docs/howto/configure/update-env.md:18]:
```suggestion
3. Get the latest tag from the [tags section](https://quay.io/repository/2i2c/2i2c-hubs-image?tab=tags) in quay.io
```

**Review by choldgraf — APPROVED** [review]
Just a few minor changes to the name - looks good to me other than that, thanks @GeorgianaElena !

**Comment by GeorgianaElena** [comment]
Thanks everyone for their feedback. I plan to merge this now :rocket:

**Pull request merged**